src/Compiler/Scheme/Racket.idr

```diff
@@ -455,23 +450,14 @@ compileExpr mkexec c s tmpDir outputDir tm outfile
          raco <- coreLift findRacoExe
          racket <- coreLift findRacket
 
-         ok <- the (Core Int) $ if mkexec
-                  then logTime 1 "Build racket" $
-                         coreLift $ system $ raco ++ ["-o", outBinAbs, outRktAbs]
-                  else pure 0
-         if ok == 0
-            then do -- TODO: add launcher script
-                    let outShRel = outputDir </> outfile
-                    if isWindows
-                       then if mkexec
-                               then makeShWindows "" outShRel appDirRel outBinFile
-                               else makeShWindows (racket ++ " ") outShRel appDirRel outRktFile
-                       else if mkexec
-                               then makeSh "" outShRel appDirRel outBinFile
-                               else makeSh (racket ++ " ") outShRel appDirRel outRktFile
-                    coreLift_ $ chmodRaw outShRel 0o755
-                    pure (Just outShRel)
-            else pure Nothing
+         when mkexec $ logTime 1 "Build racket" $
+             safeSystem $ raco ++ ["-o", outBinAbs, outRktAbs]
+
+         -- TODO: add launcher script
+         let outShRel = outputDir </> outfile
+         makeShPlatform isWindows (if mkexec then "" else racket ++ " ") outShRel appDirRel outRktFile
+         handleFileError outShRel $ chmodRaw outShRel 0o755
+         pure (Just outShRel)
```

Comment:
Again we're also dropping the `Nothing` branch

Reply:
I replaced all `Nothing` with `throw`. Now `Nothing` just wraps around to `CompilationFailed`. A more specific cause of the error is passed to `throw`.
https://github.com/idris-lang/Idris2/blob/ec74792a49bf4d22509172d1f03d153ffca1b95c/src/Idris/REPL.idr#L816